src/RootIO.jl

```diff
@@ -2,21 +2,106 @@ module RootIO
 
 import ROOT, Tables, CxxWrap
 
+"""
+    struct TTree
+
+A struct representing a ROOT TTree with its associated branches and file.
+
+# Fields
+- `_ROOT_ttree`: The ROOT TTree object.
+- `_branch_array`: An array of branches associated with the TTree.
+- `_file`: A pointer to the ROOT file where the TTree is stored.
+"""
 struct TTree
     _ROOT_ttree::ROOT.TTree
     _branch_array
+    _branch_names
     _file::CxxWrap.CxxWrapCore.CxxPtr{ROOT.TFile}
 end
 
+"""
+    Write(tree::TTree)
+
+Writes a ROOT TTree to the associated ROOT file.
+
+# Arguments
+- `tree::TTree`: The TTree object to be written to the ROOT file.
+"""
 function Write(tree::TTree)
     ROOT.Write(tree._ROOT_ttree)
 end
 
-function TTree(file::CxxWrap.CxxWrapCore.CxxPtr{ROOT.TFile}, name::String, title::String, data)
 
-    println("Creating a new TTree")
+"""
+    _makeTTree(file::CxxWrap.CxxWrapCore.CxxPtr{ROOT.TFile}, name::String, title::String, _branch_types, _branch_names)
+
+Creates a ROOT TTree with specified branches.
 
+# Arguments
+- `file`: A pointer to a ROOT file where the TTree will be stored.
+- `name`: The name of the TTree.
+- `title`: The title of the TTree.
+- `_branch_types`: A collection of types for the branches.
+- `_branch_names`: A collection of names for the branches.
+
+# Returns
+- A RootIO `TTree` object containing the ROOT TTree and its branches.
+"""
+function _makeTTree(file::CxxWrap.CxxWrapCore.CxxPtr{ROOT.TFile}, name::String, title::String, _branch_types, _branch_names)
```

Comment:
no need for underscores for the arguments:
-  _branch_types → branch_types
- _ branch_names → branch_names